controllers/types.go

```diff
@@ -27,28 +27,59 @@ type NEGStatus struct {
 
 // AutonegConfig specifies the intended configuration of autoneg
 // stored in the anthos.cft.dev/autoneg annotation
-type AutonegConfig struct {
+type OldAutonegConfig struct {
 	Name string  `json:"name"`
 	Rate float64 `json:"max_rate_per_endpoint"`
 }
 
+// AutonegConfig specifies the intended configuration of autoneg
+// stored in the controller.autoneg.dev/neg annotation
+type AutonegConfig struct {
+	BackendServices map[string]map[string]AutonegNEGConfig `json:"backend_services"`
+}
+
+type AutonegConfigTemp struct {
+	BackendServices map[string][]AutonegNEGConfig `json:"backend_services"`
+}
+
+// AutonegConfig specifies the intended configuration of autoneg
+// stored in the controller.autoneg.dev/neg annotation
+type AutonegNEGConfig struct {
+	Name        string  `json:"name,omitempty"`
+	Region      string  `json:"region,omitempty"`
+	Rate        float64 `json:"max_rate_per_endpoint,omitempty"`
+	Connections float64 `json:"max_connections_per_endpoint,omitempty"`
+}
+
 // AutonegStatus specifies the reconciled status of autoneg
 // stored in the anthos.cft.dev/autoneg-status annotation
+type OldAutonegStatus struct {
+	OldAutonegConfig
+	NEGStatus
+}
+
+// AutonegStatus specifies the reconciled status of autoneg
+// stored in the controller.autoneg.dev/neg annotation
 type AutonegStatus struct {
 	AutonegConfig
 	NEGStatus
 }
 
 // Statuses represents the autoneg-relevant structs fetched from annotations
 type Statuses struct {
-	anConfig  AutonegConfig
-	anStatus  AutonegStatus
+	anConfig  OldAutonegConfig
```

Comment:
I'd prefer that we also rename the variable names. Maybe we should just drop the "an" and use config and oldConfig?

Reply:
Dropped an and nan prefixes and went with oldConfig/config/oldStatus/status.